packages/core/admin/admin/src/App.tsx

```diff
@@ -137,7 +137,13 @@ export const App = ({ authLogo, menuLogo, showReleaseNotification, showTutorials
     [uuid, telemetryPropertiesQuery.data]
   );
 
-  if (initQuery.isLoading) {
+  /**
+   * `routes` will only be an array if EE mode is activated,
+   * therefore we need to ensure the lazy loading of EE routes
+   * has been complete otherwise the SSO repsonse URL will not be
+   * caught by the component and you won't be able to login with it.
+   */
+  if (initQuery.isLoading || (Array.isArray(routes) && routes.length === 0)) {
```

Comment:
Is there any case where routes could be 0 on purpose? Like if all ee features are disabled but it's still EE?

Reply:
We could make `useEnterprise` better for sure e.g. adding "state" like `isLoading` etc.

I think if we wanted to do that we should revisit the hook entirely, the types aren't great etc. So it'd be another ticket 🤔

Reply:
This also isn't an issue in V5 because of how the router works there.